Added NFS Export (#132)
dell/terraform-provider-powerstore: docs/resources/nfs_export.md

---

# powerstore_nfs_export (Resource)

This resource is used to manage the nfs export entity of PowerStore Array. We can Create, Update and Delete the nfs export using this resource. We can also import an existing nfs export from PowerStore array.

## Example Usage

```terraform
/*
Copyright (c) 2025 Dell Inc., or its subsidiaries. All Rights Reserved.

Licensed under the Mozilla Public License Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://mozilla.org/MPL/2.0/

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

# Commands to run this tf file : terraform init && terraform plan && terraform apply
# Create, Update, Delete and Import is supported for this resource

resource "powerstore_nfs_export" "test1" {
  // Required

>   file_system_id = "6795320a-2186-5411-fb3c-5692f12c6aa4"

Reviewer: We should use file_system datasource here to get the ID.

Author: added the same

  name = "terraform-nfs"
  path = "/terraform-fs"

  // Optional
  anonymous_gid = -24
  anonymous_uid = -24 
  description = "nfs export"
  is_no_suid = false 
  min_security = "Sys" # Options: "Sys", "Kerberos", "Kerberos_With_Integrity", "Kerberos_With_Encryption"
  default_access = "Read_Write" # Options: "No_Access", "Read_Only", "Read_Write", "Root", "Read_Only_Root"

}
```

After the execution of above resource block, nfs export would have been created on the PowerStore array. For more information, Please check the terraform state file.

<!-- schema generated by tfplugindocs -->
## Schema

### Required

> - `file_system_id` (String) The unique identifier of the nfs export on which the NFS Export will be created. The nfs export ID can be used instead of the nfs export name. For example: 'file_system_id':'name:file_system_name'

Reviewer: We should not encourage users to use the `name:<name>` construct. This will lead to cyclic dependency issues for them (and refresh issues for us, potentially).

Reviewer: ```suggestion
- `file_system_id` (String) The unique identifier of the file system on which the NFS Export will be created.'
```

Author: updated the comment
